test/unit/behaviours/remove-image.test.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const reqres = require('hof-util-reqres');
+const Behaviour = require('../../../apps/rotm/behaviours/remove-image');
+
+describe('behaviours/remove-image', () => {
+
+  it('exports a function', () => {
+    expect(Behaviour).to.be.a('function');
+  });
+
+  class Base {
+    getValues() {}
+  }
+
+  let req;
+  let request;
+  let res;
+  let instance;
+  let RemoveImage;
+
+  beforeEach(() => {
+    request = reqres.req;
+    req = request();
+    req.sessionModel.unset = sinon.stub();
+    req.get = sinon.stub();
+    res = reqres.res();
+    RemoveImage = Behaviour(Base);
+    instance = new RemoveImage();
+  });
+
+  describe('getValues()', () => {
+    beforeEach(() => {
+      sinon.stub(Base.prototype, 'getValues').yields();
+    });
+
+    afterEach(() => {
+      Base.prototype.getValues.restore();
+    });
+
+    it('removes the image if the referrer is /image', () => {
+      req.get.withArgs('Referrer').returns('localhost:8080/image');
+
+      instance.getValues(req, res, () => {
+        req.sessionModel.unset.should.be.calledWith('image');
+        req.sessionModel.unset.should.be.calledWith('image-preview');
+        req.sessionModel.unset.should.be.calledWith('image-url');
+      });
+    });
```

Comment:
do I still need to pass in done and call done when I'm not exactly calling a callback at the end of the function?
I'm doing this now.  
`super.getValues(req, res, callback);`

The assertions seem to be working as it fails if I say unset.should.not.be.called

Reply:
do this
```
instance.getValues(req, res, () => {});
req.sessionModel.unset.should.be.calledWith('image');
req.sessionModel.unset.should.be.calledWith('image-preview');
req.sessionModel.unset.should.be.calledWith('image-url');
```

otherwise it looks like a incorrect async test

Reply:
You may need to pass `done` into `getValues` like `instance.getValues(req, res, done);`

Reply:
which is better? calling done?

Reply:
I think it's better to pass done into `.getValues`.
The way you've written the test implies the conditional code happens in the callback. 
However, the call back and calling super is just a condition of the structure of the code, and nothing to do with the test

Reply:
ok, but as it's not async
I think how you redid the code above is more clearer to me like so
```
instance.getValues(req, res, () => {});
req.sessionModel.unset.should.be.calledWith('image');
req.sessionModel.unset.should.be.calledWith('image-preview');
req.sessionModel.unset.should.be.calledWith('image-url');
```